src/utils/daos_hdlr.c

```diff
@@ -1561,6 +1562,817 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			if (daos_uuid_valid(ap->src_c_uuid)) { 
+				rc = daos_cont_open(ap->pool, ap->src_c_uuid, DAOS_COO_RW,
+					&ap->cont, src_cont_info, NULL);
+				if (rc != 0) {
+					fprintf(stderr, "failed to open source cont: %d\n", rc);
+					D_GOTO(out, rc);
+				}
+				rc = dfs_mount(ap->pool, ap->cont, O_RDWR, &daos_src_file->dfs);
+				if (rc) {
+					fprintf(stderr, "dfs mount on source failed: %s, %d\n",
+						ap->src_path, rc);
+					D_GOTO(out, rc);
+				}
+			}
+		}
+	} else {
+		/* Resolve pool, container UUIDs from src path if needed */
+        	if (ap->src_path != NULL) {
+			rc = duns_resolve_path(ap->src_path, src_dattr);
+			if (rc == 0) { 
+				ap->type = (*src_dattr).da_type;
+				uuid_copy(ap->src_p_uuid, (*src_dattr).da_puuid);
+				uuid_copy(ap->src_c_uuid, (*src_dattr).da_cuuid);
+				rc = daos_pool_connect(ap->src_p_uuid, ap->sysname, NULL,
+						DAOS_PC_RW, &ap->pool, NULL, NULL);
+				if (rc != 0) {
+					fprintf(stderr, "failed to connect to destination pool: %d\n", rc);
+					D_GOTO(out, rc);
+				}
+				rc = daos_cont_open(ap->pool, ap->src_c_uuid, DAOS_COO_RW,
+					&ap->cont, src_cont_info, NULL);
+				rc = dfs_mount(ap->pool, ap->cont, O_RDWR, &(daos_src_file->dfs));
+				if (rc) {
+					fprintf(stderr, "dfs mount on source failed: %s, %d\n",
+						ap->src_path, rc);
+					D_GOTO(out, rc);
+				}
+			} else {
+				daos_src_file->type = POSIX;
+			}
+		}
+	}
+	/* connect to destination pool/conts */
+	if (daos_uuid_valid(ap->dst_p_uuid)) {
+			rc = daos_pool_connect(ap->dst_p_uuid, ap->sysname, NULL,
+			DAOS_PC_RW, &ap->dst_pool, NULL, NULL);
+			if (rc != 0) {
+				fprintf(stderr, "failed to connect to destination pool: %d\n", rc);
+				D_GOTO(out, rc);
+			}
+		if (daos_uuid_valid(ap->dst_c_uuid)) { 
+			rc = daos_cont_open(ap->dst_pool, ap->dst_c_uuid, DAOS_COO_RW,
+				&ap->dst_cont, dst_cont_info, NULL);
+			rc = dfs_mount(ap->dst_pool, ap->dst_cont, O_RDWR, &(daos_dst_file->dfs));
+			if (rc) {
+				fprintf(stderr, "dfs mount on destination failed: %s, %d\n",
+					ap->dst_path, rc);
+				D_GOTO(out, rc);
+			}
+		}
+	} else {
+		/* Resolve pool, container UUIDs from dst path if needed */
+        	if (ap->dst_path != NULL) {
+			rc = duns_resolve_path(ap->dst_path, dst_dattr);
+			if (rc == 0) { 
+				ap->type = (*dst_dattr).da_type;
+				uuid_copy(ap->dst_p_uuid, (*dst_dattr).da_puuid);
+				uuid_copy(ap->dst_c_uuid, (*dst_dattr).da_cuuid);
+				rc = daos_pool_connect(ap->dst_p_uuid, ap->sysname, NULL,
+						DAOS_PC_RW, &ap->dst_pool, NULL, NULL);
+				if (rc != 0) {
+					fprintf(stderr, "failed to connect to destination pool: %d\n", rc);
+					D_GOTO(out, rc);
+				}
+				rc = daos_cont_open(ap->dst_pool, ap->dst_c_uuid, DAOS_COO_RW,
+					&ap->dst_cont, dst_cont_info, NULL);
+				rc = dfs_mount(ap->dst_pool, ap->dst_cont, O_RDWR, &(daos_dst_file->dfs));
+				if (rc) {
+					fprintf(stderr, "dfs mount on destination failed: %s, %d\n",
+						ap->dst_path, rc);
+					D_GOTO(out, rc);
+				}
+			} else {
```

Comment:
so if path is not NULL, but not a duns path we create it? where is the call to duns_create()?
if it's a valid path but not a UNS one, we should fail though.
the only way we should create the duns path is if it doesn't  exist. the duns_resolve returns ENOENT

Reply:
@mchaarawi not sure I follow what you mean here, if if the path is not null, and it is is not a duns path, then I assume/set it to a POSIX type